Maths/GreatestCommonFactor.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @function GreatestCommonFactor
+ * @description Determine the greatest common factor of a group of numbers.
+ * @param {Number[]} nums - An array of numbers.
+ * @return {Number} - The greatest common factor.
+ * @see https://www.mathsisfun.com/greatest-common-factor.html
+ * @example GreatestCommonFactor(12, 8) = 4
+ * @example GreatestCommonFactor(3, 6) = 3
+ * @example GreatestCommonFactor(32, 16, 12) = 4
+ */
+
+export const GCFOfTwoNumbers = (a: number, b: number): number => {
+  if (!Number.isInteger(a) || !Number.isInteger(b) || a < 0 || b < 0) {
+    throw new Error("numbers must be natural to determine factors");
+  }
+
+  while (b) {
+    [a, b] = [b, a % b]
+  }
+  return a;
+}
+
+export const GreatestCommonFactor = (nums: number[]): number => {
+  if (nums.length === 0) {
+    throw new Error("at least one number must be passed in");
+  }
+
+  let greatestCommonFactor = nums[0];
+  for (const num of nums) {
+    greatestCommonFactor = GCFOfTwoNumbers(greatestCommonFactor, num)
+  }
+  return greatestCommonFactor;
```

Comment:
Consider shortening this (the entire loop construction) to `nums.reduce(GCFOfTwoNumbers)`